src/java/htsjdk/variant/vcf/VCFHeader.java

```diff
@@ -273,29 +272,36 @@ public void loadVCFVersion() {
-            } else if ( line instanceof VCFFormatHeaderLine ) {
-                final VCFFormatHeaderLine formatLine = (VCFFormatHeaderLine)line;
-                addMetaDataMapBinding(mFormatMetaData, formatLine);
-            } else if ( line instanceof VCFFilterHeaderLine ) {
-                final VCFFilterHeaderLine filterLine = (VCFFilterHeaderLine)line;
-                mFilterMetaData.put(filterLine.getID(), filterLine);
-            } else if ( line instanceof VCFContigHeaderLine ) {
-                contigMetaData.add((VCFContigHeaderLine)line);
-            } else {
-                mOtherMetaData.put(line.getKey(), line);
-            }
+            addMetadataHeaderLine(line);
         }
+    }
 
-        if ( hasFormatLine(VCFConstants.GENOTYPE_LIKELIHOODS_KEY) && ! hasFormatLine(VCFConstants.GENOTYPE_PL_KEY) ) {
-            if ( GeneralUtils.DEBUG_MODE_ENABLED ) {
-                System.err.println("Found " + VCFConstants.GENOTYPE_LIKELIHOODS_KEY + " format, but no "
-                                   + VCFConstants.GENOTYPE_PL_KEY + " field.  We now only manage PL fields internally"
-                                   + " automatically adding a corresponding PL field to your VCF header");
+    /**
+     * add a single header line to the appropriate map/list depending on its type
+     */
+    private void addMetadataHeaderLine(final VCFHeaderLine line) {
+        if ( line instanceof VCFInfoHeaderLine )  {
+            final VCFInfoHeaderLine infoLine = (VCFInfoHeaderLine)line;
+            addMetaDataMapBinding(mInfoMetaData, infoLine);
+        } else if ( line instanceof VCFFormatHeaderLine ) {
+            final VCFFormatHeaderLine formatLine = (VCFFormatHeaderLine)line;
+            addMetaDataMapBinding(mFormatMetaData, formatLine);
+
+            if ( hasFormatLine(VCFConstants.GENOTYPE_LIKELIHOODS_KEY) && ! hasFormatLine(VCFConstants.GENOTYPE_PL_KEY) ) {
+                if ( GeneralUtils.DEBUG_MODE_ENABLED ) {
+                    System.err.println("Found " + VCFConstants.GENOTYPE_LIKELIHOODS_KEY + " format, but no "
+                            + VCFConstants.GENOTYPE_PL_KEY + " field.  We now only manage PL fields internally"
+                            + " automatically adding a corresponding PL field to your VCF header");
+                }
+                addMetaDataLine(new VCFFormatHeaderLine(VCFConstants.GENOTYPE_PL_KEY, VCFHeaderLineCount.G, VCFHeaderLineType.Integer, "Normalized, Phred-scaled likelihoods for genotypes as defined in the VCF specification"));
             }
-            addMetaDataLine(new VCFFormatHeaderLine(VCFConstants.GENOTYPE_PL_KEY, VCFHeaderLineCount.G, VCFHeaderLineType.Integer, "Normalized, Phred-scaled likelihoods for genotypes as defined in the VCF specification"));
+        } else if ( line instanceof VCFFilterHeaderLine ) {
+            final VCFFilterHeaderLine filterLine = (VCFFilterHeaderLine)line;
+            mFilterMetaData.put(filterLine.getID(), filterLine);
```

Comment:
These are both good valid points.  I don't know what the intention was initially around these choices, but I agree it is probably the most consistent choice is do what addMetaDataMapBinding does and don't replace but show a message if we are in debug mode.  I'd also be ok with replacing them, but presumably there was a reason for how addMetaDataMapBinding works...